msprime/mutations.py

```diff
@@ -1171,15 +1253,24 @@ def mutate(
         end_time = float(end_time)
     if start_time > end_time:
         raise ValueError("start_time must be <= end_time")
-    keep = bool(keep)
-    discrete = bool(discrete)
-    kept_mutations_before_end_time = bool(kept_mutations_before_end_time)
+    if keep is None:
```

Comment:
What's the reason to have the default be `None` instead of `False`? It's easier to tell from the signature what the default is if it is `False`.

Reply:
It's a general principle: unless you are absolutely sure that under all possible combinations of input paraters in the future the default will be False, it's better to use a sentinel value so you can always tell whether the user specifically asked for False, or if it was the default.